### DISCUSSION TITLE
How `defineAsyncComponent`?

### DISCUSSION THREAD

**Discussion by cawa-93** [nuxt/nuxt#17486] [Questions]
How to `defineAsyncComponent` components in nuxt with autoimport? It would be useful to introduce some examples for this. The only option I see is not to use auto-import, and import the consumable component manually

```vue
<script setup>
const MyAsyncComponent = defineAsyncComponent(() => import('full/path/to/components/MyComponent.vue'))
</script>

<template>
<my-async-component v-if="condition"/>
</template>
```
Is this the right way?

**Comment by danya-rnd** [comment]
Nuxt imports all your componentы asynchronously. U can be sure of this by check the browser network, all components are separated to chunks.

The way u described between simple import and lazy. There is only one solution - disable autoimports in nuxt.config and import all components manually.

If u mean that u want to render it on client side you should use this way

```vue
<template>
  <ClientOnly>
    //your code here
  </ClientOnly>
</template>
```

**Reply by cawa-93** [reply]
My goal is to reduce the time of the first render. I would like nuxt to load components only for scratches when they are really rendered
